### PR TITLE
Tx results too large2

### PR DESCRIPTION
**Pull request by bjartek**
duplicate of https://github.com/onflow/flow-cli/pull/1118 to get all the data.

### CODE REVIEW

File: flowkit/gateway/emulator.go [hunk 1]
```diff
@@ -148,6 +148,11 @@ func (g *EmulatorGateway) GetTransactionResultsByBlockID(_ flow.Identifier) ([]*
 	panic("GetTransactionResultsByBlockID not implemented")
 }
 
+func (g *EmulatorGateway) GetTransactionResultByIndex(_ flow.Identifier, _ uint32) (*flow.TransactionResult, error) {
+	// TODO: implement
```
Comment (chasefleming) [inline flowkit/gateway/emulator.go:152]: Should this be implemented before merge?

**Review by sideninja — COMMENTED** [review]
Left some comments

File: flowkit/gateway/emulator.go (hunk 1 above)
Comment (bjartek) [reply flowkit/gateway/emulator.go:152]: There are plenty of other operations in EmulatorGateway that is not implemented. I am not even sure how to do it.

File: flowkit/flowkit.go [hunk 2]
```diff
@@ -864,16 +864,29 @@ func (f *Flowkit) GetTransactionsByBlockID(
...
+		txRes := []*flow.TransactionResult{}
+		for index := range tx {
+			txr, err := f.gateway.GetTransactionResultByIndex(blockID, uint32(index))
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "failed getting result for index %d", index)
+			}
+			txRes = append(txRes, txr)
+		}
+		return tx, txRes, nil
 	}
```
[the remark this answers is not in the document]
Comment (bjartek) [reply flowkit/flowkit.go:888]: We have no way of knowing what transactions to iterate over in that case, we could fetch the block and the collections again, but personally i do not see that it is worth it.

[the remark this answers is not in the document]
Comment (sideninja) [reply flowkit/flowkit.go:888]: Why? If you put the above added logic inside the `GetTransactionResultsByBlockID` you should be able to call then the `GetTransactionResultByIndex` by that same block ID, no?

**Comment by sideninja** [comment]
I'm closing this as discussed with @bjartek, since the increase of the limit wtih PR https://github.com/onflow/flow-cli/pull/1150 will unblock the issues we are currently facing. If there are similar issues in the future I feel streaming APIs will help getting huge data. The issue with this implementation is also it might open up other issues such as rate limiting on the ANs, since there will be a lot of sub requests to avoid one big request.

**Pull request closed without merging**